source/FreeRTOS_IP.c

```diff
@@ -798,34 +799,37 @@ void * FreeRTOS_GetUDPPayloadBuffer( size_t uxRequestedSizeBytes,
         uxBlockTime = ipconfigUDP_MAX_SEND_BLOCK_TIME_TICKS;
     }
 
-    /* Obtain a network buffer with the required amount of storage. */
-    pxNetworkBuffer = pxGetNetworkBufferWithDescriptor( sizeof( UDPPacket_t ) + uxRequestedSizeBytes, uxBlockTime );
+    if( ucIPType == ipTYPE_IPv6 )
+    {
+        uxPayloadOffset = sizeof( UDPPacket_IPv6_t );
+    }
+    else if( ucIPType == ipTYPE_IPv4 )
+    {
+        uxPayloadOffset = sizeof( UDPPacket_t );
+    }
+    else
+    {
+        /* Shouldn't reach here. */
+        configASSERT( ( ucIPType == ipTYPE_IPv6 ) || ( ucIPType == ipTYPE_IPv4 ) );
+    }
 
-    if( pxNetworkBuffer != NULL )
+    if( uxPayloadOffset != 0 )
     {
-        if( uxIPHeaderSizePacket( pxNetworkBuffer ) == ipSIZE_OF_IPv6_HEADER ) /*TODO seems incorrect */
-        {
-            uint8_t * pucIPType;
-            size_t uxIPHeaderLength;
+        /* Obtain a network buffer with the required amount of storage. */
+        pxNetworkBuffer = pxGetNetworkBufferWithDescriptor( uxPayloadOffset + uxRequestedSizeBytes, uxBlockTime );
 
-            /* Calculate the distance between the beginning of
-             * UDP payload until the hidden byte that reflects
-             * the IP-type: either ipTYPE_IPv4 or ipTYPE_IPv6.
-             */
+        if( pxNetworkBuffer != 0 )
```

Comment:
Please suggest why we are not checking against NULL. This should be if ( pxNetworkBuffer != NULL )

Reply:
Fixed, thanks